### DISCUSSION TITLE
Building on M1 Mac

### DISCUSSION THREAD

**Discussion by ad-chaos** [kovidgoyal/kitty#6482] [General]
I see a couple issues where people are unable to build on M1 Macs because brew puts the installed libraries in non-standard paths, putting something like the following in zshrc:
```zsh
export CPATH=/opt/homebrew/include/
export LIBRARY_PATH=/opt/homebrew/lib/
```
or equivalent would work. Might be worth putting it in the building docs?

**Comment by kovidgoyal** [comment]
Well, the dependency on librsync is going to go away anyway for the next
major kitty release. I think that's the only one that currently causes
problems. Though if you do want to address this probably best to just
have setup.py check if the header is available by invoking clang and if
not to look in various "common" alternate destinations on macOS and if
the header is found to add it to cflags automatically. 

On Tue, Jul 18, 2023 at 01:23:41AM -0700, ad-chaos wrote:
> I see a couple issues where people are unable to build on M1 Macs because brew puts the installed libraries in non-standard paths, putting something like the following in zshrc:
> ```zsh
> export CPATH=/opt/homebrew/include/
> export LIBRARY_PATH=/opt/homebrew/lib/
> ```
> or equivalent would work. Might be worth putting it in the building docs?
> 
> -- 
> Reply to this email directly or view it on GitHub:
> https://github.com/kovidgoyal/kitty/discussions/6482
> You are receiving this because you are subscribed to this thread.
> 
> Message ID: ***@***.***>

-- 
_____________________________________

Dr. Kovid Goyal 
https://www.kovidgoyal.net
https://calibre-ebook.com
_____________________________________
